### PR TITLE
WIP: Feature/multiprocessing

### PR DESCRIPTION
**Pull request by grahamrow**
Major changes:

1. Replace asyncio with multiprocessing (POSIX) or multithreading (Windows)
2. Plotters now using  multiprocessing (MP) as well
3. H5 files are now handled by `H5Handlers`, drastic performance increases

Things to address:

1. Killing and restarting plot servers is unreliable
2. Unit tests can hang, sometimes as a result of (1). Causes Travis CI difficulties
3. Does windows multithreading actually work?

### CODE REVIEW

**Comment by grahamrow** [comment]
I must admit my surprise that it just runs. It should be using multithreading by default if it senses that it's not on a POSIX system. Could you come up with a simple program to really stress the H5 writers? Should be able to use 5000 averages of record length 16384 or something similarly enormous using fake data with the X6.

File: src/auspex/experiment.py [hunk 1]
```diff
@@ -491,44 +502,175 @@ def catch_ctrl_c(signum, frame):
...
 
+        # Wait for the
+        time.sleep(0.1)
+        times = {n: 0 for n in self.other_nodes}
+        dones = {n: False for n in self.other_nodes}
+        while False in dones.values():
+            time.sleep(1)
+            for n in self.other_nodes:
+                if not n.done.is_set():
+                    times[n] += 1
```
Comment (dieris) [inline src/auspex/experiment.py]: It may be good to set a maximum number of `times[n]` until it stops.

**Comment by dieris** [comment]
Confirming successful tests on windows, with plotters and writers keeping up with the settings above. Saving raw and integrated data on 2 streams, for ~2 GB.

File: src/auspex/experiment.py (hunk 1 above)
Comment (grahamrow) [reply src/auspex/experiment.py]: Done!

**Comment by grahamrow** [comment]
Okay, @dieris @matthewware @gribeill what are the current implications for merging this onto develop? Should we tag a *last_asyncio* branch to join our *last_matlab* branch? I think this deserves a notable version increment while we're at it.

**Comment by dieris** [comment]
Not getting any response from the plot server in https://github.com/BBN-Q/Auspex/blob/36b8da0733c39d6604954c6440efb96b6a2f7eee/src/auspex/experiment.py#L721

**Comment by dieris** [comment]
> Not getting any response from the plot server in
> 
> [Auspex/src/auspex/experiment.py](https://github.com/BBN-Q/Auspex/blob/36b8da0733c39d6604954c6440efb96b6a2f7eee/src/auspex/experiment.py#L721)
> 
> Line 721 in [36b8da0](/BBN-Q/Auspex/commit/36b8da0733c39d6604954c6440efb96b6a2f7eee)
> 
>  evts = dict(poller.poll(1000))

Never mind, I had not started the server... All seems OK now

**Comment by dieris** [comment]
Mixer calibration fail at https://github.com/BBN-Q/Auspex/blob/731989decd56d1259cc65565b59d35e8be0d62fe/src/auspex/experiment.py#L601 with:
`RuntimeError: threads can only be started once`
when a second sweep needs to be added to the same plot. Can we keep the thread open between runs, similar to the `leave_plot_server_open` introduced in https://github.com/BBN-Q/Auspex/pull/152?

**Comment by matthewware** [comment]
Pinging all Auspexers!  Are there any major concerns with merging this into develop?  Are there issues with Windows?  Maybe 1) and 3)?

**Comment by dieris** [comment]
I believe we still need to solve the calibration plotting issues, see my comment on 9/17

**Pull request closed without merging**